RockWeb/Blocks/Groups/GroupAttendanceDetail.ascx.cs

```diff
@@ -368,9 +326,67 @@ protected void bddlCampus_SelectionChanged( object sender, EventArgs e )
 
         protected void ppAddPerson_SelectPerson( object sender, EventArgs e )
         {
+            string template = GetAttributeValue( "LavaTemplate" );
+
             if ( ppAddPerson.PersonId.HasValue )
             {
-                AddAttendee( ppAddPerson.PersonId.Value );
+                if ( !_attendees.Any( a => a.PersonId == ppAddPerson.PersonId.Value ) )
+                {
+                    var Person = new PersonService( new RockContext() ).Get( ppAddPerson.PersonId.Value );
+                    if ( Person != null )
+                    {
+                        string addPersonAs = GetAttributeValue( "AddPersonAs" );
+                        if ( !addPersonAs.IsNullOrWhiteSpace() && addPersonAs == "Group Member" )
+                        {
+                            var rockContext = new RockContext();
+                            GroupMemberService groupMemberService = new GroupMemberService( rockContext );
+                            GroupTypeRole role = new GroupTypeRoleService( rockContext ).Get( _group.GroupType.DefaultGroupRoleId ?? 0 );
+
+                            var groupMember = new GroupMember { Id = 0 };
+                            groupMember.GroupId = _group.Id;
+
+                            // check to see if the person is already a member of the group/role
+                            var existingGroupMember = groupMemberService.GetByGroupIdAndPersonIdAndGroupRoleId(
+                                _group.Id, Person.Id, _group.GroupType.DefaultGroupRoleId ?? 0 );
+
+                            if ( existingGroupMember != null )
+                            {
+                                return;
+                            }
+
+                            groupMember.PersonId = Person.Id;
+                            groupMember.GroupRoleId = role.Id;
+                            groupMember.GroupMemberStatus = GroupMemberStatus.Active;
+
+                            if ( groupMember.Id.Equals( 0 ) )
+                            {
+                                groupMemberService.Add( groupMember );
+                            }
+
+                            rockContext.SaveChanges();
```

Comment:
Once this SaveChanges() happens, the person is added to the group -- _even if they press the cancel button_ on the add attendance screen. I know you don't have much of an easy choice here to avoid this situation.   The only things I can think of would be to defer the save by storing the 'new' people in a hidden, then performing the save on the attendance Save event... or removing any newly added people during the attendance Cancel event/click.